velox/common/memory/MappedMemory.cpp

```diff
@@ -188,26 +191,35 @@ bool MappedMemoryImpl::allocate(
     MachinePageCount numPages,
     int32_t owner,
     Allocation& out,
-    std::function<void(int64_t)> beforeAllocCB,
+    std::function<void(int64_t)> userAllocCB,
     MachinePageCount minSizeClass) {
   free(out);
 
   auto mix = allocationSize(numPages, minSizeClass);
 
   if (FLAGS_velox_use_malloc) {
-    if (beforeAllocCB) {
-      uint64_t bytesAllocated = 0;
+    uint64_t bytesToAllocate = 0;
+    if (userAllocCB != nullptr) {
       for (int32_t i = 0; i < mix.numSizes; ++i) {
         MachinePageCount numPages =
             mix.sizeCounts[i] * sizeClassSizes_[mix.sizeIndices[i]];
-        bytesAllocated += numPages * kPageSize;
+        bytesToAllocate += numPages * kPageSize;
       }
-      beforeAllocCB(bytesAllocated);
+      userAllocCB(bytesToAllocate);
     }
 
     std::vector<void*> pages;
     pages.reserve(mix.numSizes);
     for (int32_t i = 0; i < mix.numSizes; ++i) {
+      if (TestValue::enabled()) {
+        bool injectAllocFailure = false;
+        TestValue::adjust(
+            "facebook::velox::memory::MappedMemoryImpl::allocate",
+            &injectAllocFailure);
```

Comment:
This name is a bit confusing because there is no obvious failure happening. I assume letting TestValue::adjust throw is not an option. Or is it?

Reply:
The actual code doesn't throw on allocation failure. I will add comment to explain the test behavior here. Thanks!